src/memory.rs

```diff
@@ -271,6 +291,26 @@ impl Memory<u16> for GlulxMemory {
     }
 }
 
+impl Memory<i16> for GlulxMemory {
+    fn read(&self, ptr: u32) -> i16 {
+        BigEndian::read_i16(&self.memory[ptr as usize..])
+    }
+
+    fn write(&mut self, ptr: u32, value: i16) {
+        BigEndian::write_i16(&mut self.memory[ptr as usize..], value)
+    }
+
+    fn ram_read(&self, ptr: u32) -> i16 {
+        let ptr = (ptr + self.ramstart()) as usize;
+        BigEndian::read_i16(&self.memory[ptr..])
+    }
+
+    fn ram_write(&mut self, ptr: u32, value: i16) {
+        let ptr = (ptr + self.ramstart()) as usize;
+        BigEndian::write_i16(&mut self.memory[ptr..], value)
+    }
+}
+
```

Comment:
actually the entire file's spacing is messed up here. That'll be fixed in it's own commit.